src/hotspot/share/prims/jvmtiThreadState.hpp

```diff
@@ -250,6 +252,13 @@ class JvmtiThreadState : public CHeapObj<mtInternal> {
 
   static void periodic_clean_up();
 
+  // JvmtiThreadState objects for virtual thread filtered events enabled globally
+  // must be created eagerly if the interp_only_mode is enabled. Otherwise,
+  // it is an important optimization to create JvmtiThreadState objects lazily.
```

Comment:
No need for this comment here. It is already at the call site, which is where it belongs. Instead the comment here should say what this API does (return true if any thread has entered interp_only_mode at any point during the JVMs execution).

Reply:
Thanks, good suggestion.

Reply:
Fixed.